src/Umbraco.Web.BackOffice/Filters/EnsureUserPermissionForMediaAttribute.cs

```diff
@@ -0,0 +1,181 @@
+﻿using System;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Umbraco.Core;
+using Umbraco.Core.Models;
+using Umbraco.Core.Services;
+using Umbraco.Web.BackOffice.Controllers;
+using Umbraco.Web.Common.Exceptions;
+using Umbraco.Web.Editors;
+using Umbraco.Web.Security;
+
+namespace Umbraco.Web.WebApi.Filters
+{
+    /// <summary>
+    /// Auth filter to check if the current user has access to the content item
+    /// </summary>
+    /// <remarks>
+    /// Since media doesn't have permissions, this simply checks start node access
+    /// </remarks>
+    internal sealed class EnsureUserPermissionForMediaAttribute : TypeFilterAttribute
+    {
+        public EnsureUserPermissionForMediaAttribute(int nodeId)
```

Comment:
I don't think it is required. If the input is a parameter, we use the the overload taking a parameter, and that handle UDIs. 
The overload using int is only for known nodes, like the RecycleBin. In that case I think int is okay.

No matter what, It was not part of the attributes I migrated, so if we need it, it could be in its own PR. UDIs are not used internally.